### PR TITLE
obrientimothya/c06-serverless01

### PR DESCRIPTION
**Pull request by obrientimothya**
# Description

Exercise (e.g. C01-E01): C06-SERVERLESS01

Student Name: Timothy O'Brien

Added all required files as solution to C06-SERVERLESS01

## Type of change

- [ ] Exercise Submission (one exercise per PR)
  - [ ] This PR is just for **one** exercise of a class
  - [ ] PR name follows the pattern `<github-username>/<exercise-number>`
  - [ ] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/06class/exercises/c06-serverless01/obrientimothya/Makefile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+DOCKER=docker-compose run --rm
```
Comment (denstorti) [inline classes/06class/exercises/c06-serverless01/obrientimothya/Makefile:1]: Well done, 3M. I could run it without any problems 👍

File: classes/06class/exercises/c06-serverless01/obrientimothya/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- [_provider.tf](_provider.tf)
+- [main.tf](main.tf)
+- [output.tf](output.tf)
+- [Makefile](Makefile)
+- [output.md](output.md)
+
+## Command Execution Output
+
+```
+$ curl -v \
```
Comment (denstorti) [inline classes/06class/exercises/c06-serverless01/obrientimothya/README.md:14]: good example

File: classes/06class/exercises/c06-serverless01/obrientimothya/main.tf [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  source_code_hash = filebase64sha256(local.lambda_payload)
+  function_name    = local.app_name
+  role             = aws_iam_role.lambda.arn
+
+  runtime          = local.lambda_runtime
+  handler          = local.lambda_handler
+
+  environment {
+    variables = {
+      DB_HOST = data.aws_ssm_parameter.db_host.value
```
Comment (denstorti) [inline classes/06class/exercises/c06-serverless01/obrientimothya/main.tf:183]: The env var used by the Lambda is DB_NAME (as per the exercise description), but it defaults to `DA_Serverless`. That's why the solution is working even if the value is not read from SSM.

Minor fix needed

**Review by denstorti — CHANGES_REQUESTED** [review]
Good work, Tim. Minor fix needed in the env var DB_NAME.

File: classes/06class/exercises/c06-serverless01/obrientimothya/main.tf (hunk 3 above)
Comment (obrientimothya) [reply classes/06class/exercises/c06-serverless01/obrientimothya/main.tf]: Oops! I've changed them all to `DB_NAME` now :-) in commit e752f608be753ad07c411ee57e9ab089d51375b0
Thanks, @denstorti

**Review by denstorti — APPROVED** [review]
Thanks, Tim. Looks good.

**Pull request merged**